src/components/ShareButton/ShareButton.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ShareIcon from '@mui/icons-material/Share';
+import Button from '@mui/material/Button';
+
+import { Share, StyledMenu, StyledMenuItem } from './styled';
+
+const ShareButton = ({
+  handleShare,
+  handleClick,
+  handleClose,
+  open,
+  anchorEl,
+  options,
+}) => (
+  <>
+    <Button>
+      <div data-testid="share_button">
+        <ShareIcon
+          aria-label="more"
+          id="long-button"
+          aria-controls={open ? 'long-menu' : undefined}
+          aria-expanded={open ? 'true' : undefined}
+          aria-haspopup="true"
+          onClick={handleClick}
+        />
+        <StyledMenu
+          id="long-menu"
+          anchorEl={anchorEl}
+          open={open}
+          onClose={handleClose}
+        >
+          {options.map(option => (
+            <StyledMenuItem key={option} onClick={event => handleShare(event)}>
```

Comment:
No need to use an arrow function here
```suggestion
            <StyledMenuItem key={option} onClick={handleShare}>
```